src/connect.test.js

```diff
@@ -0,0 +1,79 @@
+'use strict';
+
+const connect = require('./connect');
+
+describe('Basic scenario', () => {
+  const baseParams = { x: 1, y: 2 };
+  const extraParams = { z: 3, x: 10 };
+  let connected;
+
+  beforeEach(() => {
+    connected = connect(baseParams);
+  });
+
+  test('should return expected object', () => {
+    expect(connected(extraParams))
+      .toStrictEqual({ x: 10, y: 2, z: 3 });
+  });
+
+  test('should not change baseParams', () => {
+    expect(baseParams)
+      .toStrictEqual({ x: 1, y: 2 });
+  });
+
+  test('should not change extraParams', () => {
+    expect(extraParams)
+      .toStrictEqual({ z: 3, x: 10 });
+  });
+});
+
+describe('Without baseParams', () => {
+  const extraParams = { x: 1, y: 2, z: 3 };
+  let connected;
+
+  beforeEach(() => {
+    connected = connect();
+  });
+
+  test('should return expected object', () => {
+    expect(connected(extraParams))
+      .toStrictEqual({ x: 1, y: 2, z: 3 });
+  });
+
+  test('should not change extraParams', () => {
+    expect(extraParams)
+      .toStrictEqual({ x: 1, y: 2, z: 3 });
```

Comment:
it would be great to check here that result is a different object than `extraParams`
you can add the second expect to this test